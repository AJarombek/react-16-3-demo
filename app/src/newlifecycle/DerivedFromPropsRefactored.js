/**
 * Refactoring of the {@code getDerivedStateFromProps()} sample component to be stateless.
 * @author Andrew Jarombek
 * @since 1/21/2020
 */

import React from 'react';
import PropTypes from 'prop-types';

const DerivedFromPropsRefactored = ({show}) => {
  return (
    <div className="derived-from-props">
      { show ?
        <p className="secret-code">
          You have a beautiful heart.
        </p>
        :
        <div className="secret-code-classified">
          <p>CLASSIFIED</p>
        </div>
      }
    </div>
  )
};

DerivedFromPropsRefactored.propTypes = {
  show: PropTypes.bool.isRequired
};

export default DerivedFromPropsRefactored;
