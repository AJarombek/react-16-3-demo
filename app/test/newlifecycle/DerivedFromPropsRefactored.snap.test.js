/**
 * Snapshot test for the {@link ../../src/newlifecycle/DerivedFromPropsRefactored} component.
 * @author Andrew Jarombek
 * @since 1/29/2020
 */

import React from 'react';
import DerivedFromPropsRefactored from '../../src/newlifecycle/DerivedFromPropsRefactored';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<DerivedFromPropsRefactored show={true}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
