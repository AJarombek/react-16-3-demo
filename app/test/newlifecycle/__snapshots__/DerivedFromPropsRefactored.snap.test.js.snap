// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="derived-from-props"
>
  <p
    className="secret-code"
  >
    You have a beautiful heart.
  </p>
</div>
`;
